Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -277,3 +277,302 @@ class GetRandomNumberTrivia extends UseCase<NumberTrivia, NoParams> {
 And now the **second** test will pass.
 
 ![Second_test_pass](https://user-images.githubusercontent.com/27420533/74590474-aa92e180-5006-11ea-8a88-ef8a9a6311e7.png)
+
+#### 4. Data Layer Overview & Models
+> Video 4: https://www.youtube.com/watch?v=keaTZ9M_U1A&t=1516s
+
+In the `fourth` tutorial start by giving a brief explanation of how the models work.
+
+The models are like entities but with more features in the case of this specific application will deal with making the data conversation in Json to Dart.
+
+After this explanation is necessary to create a file inside the **models** folder with the name `number_trivia_model.dart`.
+And then its test version inside the **test** folder with the name `number_trivia_model_test.dart`.
+
+This will be the code that will be inside the test file.
+
+```dart
+import 'package:clean_architecture_tdd_prep/features/number_trivia/data/models/number_trivia_model.dart';
+import 'package:clean_architecture_tdd_prep/features/number_trivia/domain/entities/number_trivia.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  final tNumberTriviaModel = NumberTriviaModel(number: 1, text: 'Test Text');
+
+  test(
+    'should be a subclass of NumberTrivia entity',
+    () async {
+      // assert
+      expect(tNumberTriviaModel, isA<NumberTrivia>());
+    },
+  );
+}
+```
+The test will **not pass** so we will go to the `number_trivia_model.dart` file and pass some parameters.
```

Comment:
```suggestion
The test will ***not pass***. This is _expected_ because we have not yet implemented the `NumberTriviaModel` class.
Open the `number_trivia_model.dart` file and add the following code:
```